src/library_syscall.js

```diff
@@ -114,10 +114,12 @@ var SyscallsLibrary = {
       }
       return 0;
     },
-    doDup: function(path, flags, suggestFD) {
+    doDup: function(stream, suggestFD) {
       var suggest = FS.getStream(suggestFD);
       if (suggest) FS.close(suggest);
-      return FS.open(path, flags, 0, suggestFD, suggestFD).fd;
+
+      FS.streams[suggestFD] = stream;
```

Comment:
I might be ok with this, but is this right? The same stream object will be shared by two ids - but they will stay the same. The open here was creating a copy.